Elevator model

pholzer opened the pull request:
Refactored the model structure, implemented the updating mechanism for the model classes and building of the model classes + unit tests

Commit:
Initial implementation of the data models

pholzer committed:
added unit tests

pholzer committed:
implemented advanced update architecture using the bridge pattern

pholzer committed:
refactored factory interfaces and added missing aggregations to the bridge pattern on the buildingupdater

pholzer committed:
fixed spelling error in ElevatorServiceException

pholzer committed:
added unit tests for ModelFactory and Building

pholzer committed:
added building updater unittests

pholzer committed:
added elevator updater unittests

pholzer committed:
added floor updater unittests

pholzer committed:
added UpdaterFactory unittests

pholzer committed:
added RMIElevatorSerivce unittests

pholzer committed:
fixed failing unittests

pholzer committed:
eliminated some code smells

pholzer committed:
code smells

pholzer committed:
still smells

ProteinBreiti commented on src/main/java/at/fhhagenberg/model/Building.java:
```diff
@@ -0,0 +1,168 @@
+
+    private final List<Floor> mFloors;
+
+    private final boolean[][] mElevatorToFloorServiceMapping;
```
kommentar wäre nice gewesen. bin ma ned sicher was des macht

ProteinBreiti replied:
hab mi scho erinnert was des is. sorry for the inconvenience. have a nice day god bless you bye

ProteinBreiti commented on src/main/java/at/fhhagenberg/updater/BuildingUpdater.java:
```diff
@@ -0,0 +1,82 @@
+     * Updates the Building object.
+     */
+    private void updateBuilding() {
+        //TODO(PH): discuss if it is necessary to do that every update cycle
```
I würd schon sagen

Commit:
Moved all value checks into the setters, so models can only store valid values. the updaters just forward them to the models.
And moved some getters from the setter region into the getter region to protect those suffering from ocd

Commit:
Added ElevatorTest since some logic is contained in Elevator.

Commit:
Removed public keywords from test functions

Commit:
Forgot to remove public of the class

Commit:
Added @Test for some test cases where i forgot it

Commit:
Did a copy paste oopsy

Commit:
removed todos
